retrieval_service/datastore/providers/postgres.py

```diff
@@ -364,5 +383,104 @@ async def search_flights_by_airports(
         results = [models.Flight.model_validate(dict(r)) for r in results]
         return results
 
+    async def validate_ticket(
+        self,
+        airline: str,
+        flight_number: str,
+        departure_airport: str,
+        arrival_airport: str,
+        departure_time: datetime,
+        arrival_time: datetime,
+    ) -> bool:
+        results = await self.__pool.fetch(
+            """
+                SELECT * FROM flights
+                WHERE airline ILIKE $1
+                AND flight_number ILIKE $2
+                AND departure_airport ILIKE $3
+                AND arrival_airport ILIKE $4
+                AND departure_time = $5::timestamp
+                AND arrival_time = $6::timestamp;
+            """,
+            airline,
+            flight_number,
+            departure_airport,
+            arrival_airport,
+            departure_time,
+            arrival_time,
+            timeout=10,
+        )
+        if len(results) == 1:
+            return True
+        return False
+
+    async def insert_ticket(
+        self,
+        user_id: str,
+        user_name: str,
+        user_email: str,
+        airline: str,
+        flight_number: str,
+        departure_airport: str,
+        arrival_airport: str,
+        departure_time: str,
+        arrival_time: str,
+    ):
+        departure_time_datetime = datetime.strptime(departure_time, "%Y-%m-%d %H:%M:%S")
+        arrival_time_datetime = datetime.strptime(arrival_time, "%Y-%m-%d %H:%M:%S")
+        if not await self.validate_ticket(
+            airline,
+            flight_number,
+            departure_airport,
+            arrival_airport,
+            departure_time_datetime,
+            arrival_time_datetime,
+        ):
+            raise Exception("Flight information not in database")
+        results = await self.__pool.execute(
+            """
+                INSERT INTO tickets (
+                    user_id,
+                    user_name,
+                    user_email,
+                    airline,
+                    flight_number,
+                    departure_airport,
+                    arrival_airport,
+                    departure_time,
+                    arrival_time
+                ) VALUES (
+                   $1, $2, $3, $4, $5, $6, $7, $8, $9
+                );
+            """,
+            user_id,
+            user_name,
+            user_email,
+            airline,
+            flight_number,
+            departure_airport,
+            arrival_airport,
+            departure_time_datetime,
+            arrival_time_datetime,
+            timeout=10,
+        )
+        if results != "INSERT 0 1":
+            raise Exception("Ticket Insertion failure")
```

Comment:
Shouldn't it throw an error if the insert wasn't successful already?

Reply:
Good point... deleted this line